app/src/org/odk/collect/android/utilities/NepaliDateUtilities.java

```diff
@@ -0,0 +1,297 @@
+package org.odk.collect.android.utilities;
+
+import static org.odk.collect.android.utilities.UniversalDate.MILLIS_IN_DAY;
+import android.util.SparseArray;
+
+/**
+ * Date manipulation utility functions for the Nepali calendar system.
+ * 
+ * @author Richard Lu
+ */
+public class NepaliDateUtilities {
+    
+    private static final SparseArray<int[]> NEPALI_YEAR_MONTHS = new SparseArray<int[]>();
+    
+    private static final int MIN_YEAR = 2000;
+    
+    /*
+     * Nepali calendar system has no discernible cyclic month pattern, so we must manually
+     * enter them here as new calendars are known.
+     * 
+     * TODO: Enter month lengths for years beyond 2090
+     */
+    static {
+        NEPALI_YEAR_MONTHS.put(2000, new int[] { 0, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 29, 31 });
```

Comment:
2000 is in Nepali years, so it's actually 1944. I did just now find one online which goes back to 1969 BS (1912/13 AD), so might be useful to add back to then